core/src/com/unciv/models/ruleset/Building.kt

```diff
@@ -224,6 +224,8 @@ class Building : NamedStats(), IConstruction{
 
 
     override fun shouldBeDisplayed(construction: CityConstructions): Boolean {
+        if (construction.isBeingConstructed(name) || construction.isEnqueued(name))
```

Comment:
You can use `CityConstructions.isBeingConstructedOrEnqueued` here. Probably it would be good to rename the parameter `construction` to `cityConstructions` for clarity. 

As for the rest, I think this is ok :)

Reply:
Thank you for the input.